lib/task_listener.js

```diff
@@ -322,43 +322,40 @@ export default class TaskListener extends EventEmitter {
    * directly from the TC queue service, then calling runTask with the
    * resulting task set.
    */
-  async coalesceClaim(claim) {
+  async applySuperseding(claim) {
     let task = claim.task;
     let taskId = claim.status.taskId;
 
     try {
-      // if the task is not coalescible, then the taskset is just the one claim
-      if (!task.payload.coalescer) {
-        this.runtime.log('not coalescing', {taskId, message: 'no coalescer in payload'});
+      // if the task is not set up to supersede anything, then the taskset is just the one claim
+      if (!task.payload.supersederUrl) {
+        this.runtime.log('not superseding', {taskId, message: 'no supersederUrl in payload'});
         return [claim];
       }
 
-      let routes = task.routes;
-      let routePrefix = task.payload.coalescer.routePrefix + ".";
-      routes = routes.filter(r => r.startsWith(routePrefix));
-      if  (routes.length != 1) {
-        this.runtime.log('not coalescing', {taskId, message: 'no single coalescer route in payload'});
+      let supersederUrl = task.payload.supersederUrl;
+
+      let tasks = (await this.fetchSupersedingTasks(supersederUrl, taskId));
+      if (!tasks || tasks.length == 0 || tasks[0] === taskId) {
```

Comment:
I think I might be misreading something here, but I was looking at `tasks[0] === taskId` and while 
looking over the documentation here [1] it seems that you could have a response such that task[0] is the task claimed initially, but there might be task[1] and task[2] where task[2] should be the the task executed (if it can be claimed).  Should this be `tasks[tasks.length -1] === taskId` ?

[1] https://github.com/taskcluster/taskcluster-docs/pull/69/files#diff-dc8e655f527336a70d0dea380d8ec57aR405